src/wallet/coinselection.cpp

```diff
@@ -93,114 +87,127 @@ static const size_t TOTAL_TRIES = 100000;
 util::Result<SelectionResult> SelectCoinsBnB(std::vector<OutputGroup>& utxo_pool, const CAmount& selection_target, const CAmount& cost_of_change,
                                              int max_selection_weight)
 {
-    SelectionResult result(selection_target, SelectionAlgorithm::BNB);
-    CAmount curr_value = 0;
-    std::vector<size_t> curr_selection; // selected utxo indexes
-    int curr_selection_weight = 0; // sum of selected utxo weight
-
-    // Calculate curr_available_value
-    CAmount curr_available_value = 0;
+    // Check that there are sufficient funds
+    CAmount total_available = 0;
```

Comment:
Is there some rhyme or reason behind the refactor of the variable name here?  I probably would have put that into a different commit personally.  I prefer also just dropping `curr_` although I notice this is done seemingly at random since there are other variables such as `curr_selection_waste` where the original name was maintained?

Reply:
Thanks.  That makes sense since now there is the lookahead, so we don't use `curr_available_value`.  I still think dropping `curr_` from like `curr_selection` is nicer.  So there is `selection` and `best_selection`.

Also, on the topic of lookahead.  Since coin-grinder also uses lookahead, couldn't that code be shared by creating a lookahead function?  Doesn't need to happen for this PR but it would be more DRY that way.